ChatSDK/ChatSDKUI/Classes/Components/Chat View/BTextInputView.m

```diff
@@ -68,8 +67,12 @@ - (id)init
         UIBarButtonItem * sendItem = [[UIBarButtonItem alloc] initWithCustomView:_sendButton];
 
         // Create a text view
-        _textView = [[UITextView alloc] init];
-        _textView.delegate = self;
+        _textView = [[HKWTextView alloc] init];
+        
+        // If we use the mentions functionality we need to set the external delegate
+        // This is the way to set the UITextView delegate to keep mentions functionality working
+        _textView.simpleDelegate = self;
+        
```

Comment:
Is the simple delegate a super set of the standard delegate method? If we get rid of the delegate, will we still get callbacks when text is typed etc...? Have you tested the mentions functionality with multiple lines?

Reply:
The HKWTextView has two delegates, one for the advanced functionality and one for the regular textView functionality. 

This means we need to set the simpleDelegate here to ensure all the typing functions are called. (You can test this by putting a break point in shouldChangeTextInRange). Later in the mention view we call the actual delegate which then connects it to all the mention side of things. 

Setting the simple delegate here allows the HKWTextView to act as a regular textView until we add the mentions module and then it gets upgraded.